azure-proto-core-test/SystemAssignedIdentityTests.cs

```diff
@@ -0,0 +1,299 @@
+        [TestCase(1, "72f988bf-86f1-41af-91ab-2d7cd011db47", "de29bab1-49e1-4705-819b-4dfddceaaa98", null, null)]
+        [TestCase(1, "72f988bf-86f1-41af-91ab-2d7cd011db47", "de29bab1-49e1-4705-819b-4dfddceaaa98", "72f988bf-86f1-41af-91ab-2d7cd011db46", "de29bab1-49e1-4705-819b-4dfddceaaa97")]
+        [TestCase(-1, null, null, "72f988bf-86f1-41af-91ab-2d7cd011db47", "de29bab1-49e1-4705-819b-4dfddceaaa98")]
+        [TestCase(-1, "72f988bf-86f1-41af-91ab-2d7cd011db47", "de29bab1-49e1-4705-819b-4dfddceaaa98", "72f988bf-86f1-41af-91ab-2d7cd011db48", "de29bab1-49e1-4705-819b-4dfddceaaa99")]
+        public void CompareTo(int answer, string tenantId1, string principalId1, string tenantId2, string principalId2)
+        {
+            SystemAssignedIdentity identity1;
+            SystemAssignedIdentity identity2;
+            if (tenantId1 == null)
+                identity1 = new SystemAssignedIdentity();
+
+            else
+                identity1 = new SystemAssignedIdentity(new Guid(tenantId1), new Guid(principalId1));
+
+            if (tenantId2 == null)
+                identity2 = new SystemAssignedIdentity();
+
+            else
+                identity2 = new SystemAssignedIdentity(new Guid(tenantId2), new Guid(principalId2));
+
+            Assert.AreEqual(answer, identity1.CompareTo(identity2));
+        }
+
+        [TestCase(null, null, null, null)]
+        [TestCase("72f988bf-86f1-41af-91ab-2d7cd011db47", "de29bab1-49e1-4705-819b-4dfddceaaa98", "72f988bf-86f1-41af-91ab-2d7cd011db47", "de29bab1-49e1-4705-819b-4dfddceaaa98")]
+        public void EqualsMethodTrue(string tenantId1, string principalId1, string tenantId2, string principalId2)
+        {
+            SystemAssignedIdentity identity1;
+            SystemAssignedIdentity identity2;
+            if (tenantId1 == null)
+                identity1 = new SystemAssignedIdentity();
+
+            else
+                identity1 = new SystemAssignedIdentity(new Guid(tenantId1), new Guid(principalId1));
+
+            if (tenantId2 == null)
+                identity2 = new SystemAssignedIdentity();
+
+            else
+                identity2 = new SystemAssignedIdentity(new Guid(tenantId2), new Guid(principalId2));
+
+            Assert.IsTrue(identity1.Equals(identity2));
+        }
+
+        [TestCase(null, null, "72f988bf-86f1-41af-91ab-2d7cd011db47", "de29bab1-49e1-4705-819b-4dfddceaaa98")]
+        [TestCase("72f988bf-86f1-41af-91ab-2d7cd011db47", "de29bab1-49e1-4705-819b-4dfddceaaa98", "72f988bf-86f1-41af-91ab-2d7cd011db44", "de29bab1-49e1-4705-819b-4dfddceaaa94")]
+        [TestCase("72f988bf-86f1-41af-91ab-2d7cd011db47", "de29bab1-49e1-4705-819b-4dfddceaaa98", null, null)]
+        [TestCase("72f988bf-86f1-41af-91ab-2d7cd011db47", "de29bab1-49e1-4705-819b-4dfddceaaa93", "72f988bf-86f1-41af-91ab-2d7cd011db42", "de29bab1-49e1-4705-819b-4dfddceaaa91")]
```

Comment:
same comment as above.  It would be good to have a complete representation of the equivalence classes:
- Tid1 == Tid2, Pid1 != Pid2
- Tid != Tid2, Pid1 == Pid2
- Tid1 != Tid2, Pid1 != Pid2

In the above, you have good coverage of the possible nulls, but have two examples of both values being unequal, rather than mixing one element equal and one element unequal variations